src/Bookings.js

```diff
@@ -1,21 +1,155 @@
+  const [buttonId, setButtonId] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
   const search = searchVal => {
     console.info("TO DO!", searchVal);
+    let filteredBookings = bookings.filter(booking => {
+      return (
+        booking.firstName.toUpperCase().includes(searchVal.toUpperCase()) ||
+        booking.surname.toUpperCase().includes(searchVal.toUpperCase())
+      );
+    });
+    setBookings(filteredBookings);
   };
+  useEffect(() => {
+    fetch("https://cyf-react.glitch.me/delayed")
+      .then(resp => {
+        if (!resp.ok) {
+          throw new Error(resp.status);
+        } else {
+          return resp.json();
+        }
+      })
+      .then(data => {
+        setBookings(data);
+        setLoading(false);
+      })
+      .catch(error => {
+        console.log(error);
+        setError(error.message);
+      });
+  }, []);
+  function changeButtonId(e) {
+    setButtonId(e.target.id);
+  }
+  const [columnClicked, setColumnClicked] = useState([
+    false,
+    false,
+    false,
+    false
+  ]);
 
-  return (
-    <div className="App-content">
-      <div className="container">
-        <Search search={search} />
-        {/* <SearchResults results={FakeBookings} /> */}
+  function columnSorter(e) {
+    let id = parseInt(e.target.id);
+    if (id === 0) {
+      if (columnClicked[0] === false) {
+        setColumnClicked([true, false, false, false]);
+        let newBookings = bookings.sort((book1, book2) => {
+          return book2.id - book1.id;
+        });
+        setBookings(newBookings);
+      } else {
+        setColumnClicked([false, false, false, false]);
+        let newBookings = bookings.sort((book1, book2) => {
+          return book1.id - book2.id;
+        });
+        setBookings(newBookings);
+      }
+    } else if (id === 1) {
+      if (columnClicked[1] === false) {
+        setColumnClicked([false, true, false, false]);
+        let newBookings = bookings.sort((book1, book2) => {
+          return book1.firstName.localeCompare(book2.firstName);
+        });
+        setBookings(newBookings);
+      } else {
+        setColumnClicked([false, false, false, false]);
+        let newBookings = bookings.sort((book1, book2) => {
+          return book2.firstName.localeCompare(book1.firstName);
+        });
+        setBookings(newBookings);
+      }
+    } else if (id === 2) {
+      setColumnClicked([false, false, true, false]);
+      let newBookings = bookings.sort((book1, book2) => {
+        return book1.surname.localeCompare(book2.surname);
+      });
+      newBookings = [...newBookings];
+      setBookings(newBookings);
+    } else if (id === 3) {
+      setColumnClicked([false, false, false, true]);
+      let newBookings = bookings.sort((book1, book2) => {
+        return book1.email.localeCompare(book2.email);
+      });
+      newBookings = [...newBookings];
+      setBookings(newBookings);
```

Comment:
You're writing `setBookings` inside every `if` statement; why not just write it once at the end?

Also, this is a bit harder, but can you think of a way to only write `bookings.sort` once instead of inside every `if` statement?